precompiles/gov/gov.go

```diff
@@ -94,48 +94,50 @@ func (p Precompile) Run(evm *vm.EVM, contract *vm.Contract, readOnly bool) (bz [
 
 	// This handles any out of gas errors that may occur during the execution of a precompile tx or query.
 	// It avoids panics and returns the out of gas error so the EVM can continue gracefully.
-	defer cmn.HandleGasError(ctx, contract, initialGas, &err)()
+	defer cmn.HandleGasError(ctx, contract, initialGas, &err, stateDB, snapshot)()
 
 	if err := stateDB.Commit(); err != nil {
 		return nil, err
```

Comment:
@MalteHerrmann can you verify that doing a commit here is correct? It's not present on main